### PR TITLE
adding persistent template test and updating some

### PR DESCRIPTION
**Pull request by jitendar-singh**
- Adding test for persistent template
- Adding scenario to tear down the test project for all resources created while testing the ephemeral template
- Adding maven agent test for persistent template
- Adding nodejs agent test for persistent template

### CODE REVIEW

File: smoke/features/persistent-jenkins.feature [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Feature: Deploy Jenkins with persistent volume on openshift using template based install
```
Comment (gabemontero) [inline smoke/features/persistent-jenkins.feature:1]: So best as I can tell, this test assumes the persistent volumes are created prior to the invocation of the test.

If that is correct @jitendar-singh I believe that should be explicitly called out here and in the other persistent*.feature files.

Comment (gabemontero) [reply smoke/features/persistent-jenkins.feature:1]: Or, ideally, you replicate the NFS based PV setup we do in the current extended tests at https://github.com/openshift/origin/blob/master/test/extended/builds/pipeline_jenkins_e2e.go#L60-L72

Then we could be on a path of removing those persistent template tests in openshift/origin and replace them with CI invocations of your new smoke tests.

**Review by gabemontero — COMMENTED** [review]
Aside from the PV related suggestions below @jitendar-singh 

- outside of this PR, but ideally sooner rather than later, if you are not already on this path, I'd love to see all this cool stuff you are doing invoked from CI for any PRs against this repo
- I could also see use defining periodic / batch jobs of the smoke tests against various versions of openshift
- a few pointers from what we do in other repos that could help get you started
https://github.com/openshift/release/blob/master/ci-operator/config/openshift/builder/openshift-builder-master.yaml#L52-L59
https://github.com/openshift/release/blob/master/ci-operator/config/openshift/csi-driver-projected-resource/openshift-csi-driver-projected-resource-master.yaml#L44-L65

and add analogous entries for the smoke tests to

https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/jenkins/openshift-jenkins-master-presubmits.yaml
and
https://github.com/openshift/release/blob/master/ci-operator/config/openshift/jenkins/openshift-jenkins-master.yaml

Otherwise, I'm certainly not fluent enough in python to offer you stylistic / detailed comments on that actual code.  To some extent, being able to run them from CI with repetition and consistency would be the true validation for me.

File: smoke/features/persistent-jenkins.feature (hunk 1 above)
Comment (jitendar-singh) [reply smoke/features/persistent-jenkins.feature:1]: Thanks @gabemontero I will work this next.

**Pull request merged**